tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+            from,
+            into,
+        };
+        let subquery_name = subquery_decl.fqsqn(&helper.module);
+        helper
+            .subqueries
+            .insert(subquery_name, subquery_decl.clone());
+        Ok(subquery_decl)
+    }
+}
+
+/// we're forced to make this pub because of lalrpop
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct SubqueryStmtRaw<'script> {
+    pub(crate) start: Location,
+    pub(crate) end: Location,
+    pub(crate) id: String,
+    pub(crate) target: String,
+    pub(crate) module: Vec<IdentRaw<'script>>,
+    pub(crate) params: MaybeParams<'script>,
+}
+impl_expr!(SubqueryStmtRaw);
+
+impl<'script> SubqueryStmtRaw<'script> {
+    fn mangle_id(&self, id: &str) -> String {
+        //TTODO Remove `A`
+        format! {"A__SUBQ__{}_{}",self.id, id}
+    }
+
+    fn get_args_map<'registry>(
+        &self,
+        decl_params: &Option<HashMap<String, Value<'script>>>,
+        stmt_params: Option<HashMap<String, Value<'script>>>,
+        helper: &Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match &decl_params {
+            // No params in the declaration
+            None => Ok(HashMap::new()),
+            Some(decl_params) => {
+                match stmt_params {
+                    // No params in the creation
+                    None => Ok(decl_params.clone()),
+                    // Merge elided params from the declaration.
+                    Some(mut stmt_params) => {
+                        for (param, value) in decl_params {
+                            if !stmt_params.contains_key(param) {
+                                stmt_params.insert(param.clone(), value.clone());
+                            }
+                        }
+                        // Make sure no new params were introduced in creation
+                        if stmt_params.len() == decl_params.len() {
+                            Ok(stmt_params)
+                        } else {
+                            error_generic(self, self, &"Unknown parameter", &helper.meta)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn inline_params<'registry>(
+        params: Params<'script>,
+        sq_args: &HashMap<String, Value<'script>>,
+        helper: &mut Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match params {
+            // ALLOW: Only Raw params can be inlined.
+            Params::Processed(_) => unreachable!("Can't inline processed params."),
+            Params::Raw(params) => params
+                .into_iter()
+                .map(|(name, value)| {
+                    let value = value.up(helper)?;
+                    if let ImutExprInt::Path(Path::Reserved(ReservedPath::Args {
+                        segments, ..
+                    })) = &value
+                    {
+                        if let Segment::Id { key, .. } = &segments[0] {
```

Comment:
I'm not sure about this, it seems to assume that the form is always `args.<key>` how do we guarantee that? could someone use `args.snot.badger` here? if so `badger` would be ignored. if someone uses `args` this would also crash the runtime.